compiler/rustc_middle/src/mir/coverage.rs

```diff
@@ -226,12 +278,47 @@ pub struct BranchInfo {
     /// data structures without having to scan the entire body first.
     pub num_block_markers: usize,
     pub branch_spans: Vec<BranchSpan>,
+    pub decision_spans: Vec<DecisionSpan>,
 }
 
 #[derive(Clone, Debug)]
 #[derive(TyEncodable, TyDecodable, Hash, HashStable, TypeFoldable, TypeVisitable)]
 pub struct BranchSpan {
     pub span: Span,
+    pub condition_info: ConditionInfo,
```

Comment:
I don't think we should store MC/DC information in the existing `BranchSpan` struct.

Instead, we should have a separate list of `struct MCDCBranchSpan`.

That way, normal branch info and MC/DC branch info can be kept completely separate.